egs/tedlium/s5/local/run_cleanup_segmentation.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+# Copyright 2016  Vimal Manohar
+# Apache 2.0
+
+# This script demonstrates how to re-segment training data selecting only the 
+# "good" audio that matches the transcripts.
+# The basic idea is to decode with an existing in-domain acoustic model, and a
+# biased language model built from the reference, and then work out the
+# segmentation from a ctm like file.
+
+# For nnet3 and chain results after cleanup, see the scripts in 
+# local/nnet3/run_tdnn.sh and local/chain/run_tdnn.sh 
+
+# GMM Results:
+## %WER 26.7 | 507 17792 | 77.7 16.9 5.4 4.5 26.7 95.3 | -0.329 | exp/tri3/decode_dev.si/score_12_1.0/ctm.filt.filt.sys
+## %WER 22.0 | 507 17792 | 81.6 13.2 5.2 3.6 22.0 93.9 | -0.189 | exp/tri3/decode_dev/score_13_1.0/ctm.filt.filt.sys
+## %WER 25.5 | 1155 27512 | 77.9 17.4 4.6 3.4 25.5 92.9 | -0.270 | exp/tri3/decode_test.si/score_12_1.0/ctm.filt.filt.sys
+## %WER 20.3 | 1155 27512 | 82.7 13.4 3.9 3.0 20.3 90.0 | -0.063 | exp/tri3/decode_test/score_14_0.5/ctm.filt.filt.sys
+
+## %WER 26.2 | 507 17792 | 77.9 16.3 5.8 4.2 26.2 95.7 | -0.340 | exp/tri3_cleaned_b/decode_dev.si/score_13_1.0/ctm.filt.filt.sys
+## %WER 21.9 | 507 17792 | 81.6 13.0 5.4 3.5 21.9 93.7 | -0.147 | exp/tri3_cleaned_b/decode_dev/score_13_1.0/ctm.filt.filt.sys
+## %WER 25.2 | 1155 27512 | 78.0 17.1 4.9 3.2 25.2 92.4 | -0.257 | exp/tri3_cleaned_b/decode_test.si/score_13_1.0/ctm.filt.filt.sys
+## %WER 19.8 | 1155 27512 | 83.0 13.0 4.0 2.8 19.8 89.5 | -0.041 | exp/tri3_cleaned_b/decode_test/score_14_0.5/ctm.filt.filt.sys
+
+set -e 
+set -o pipefail 
+set -u
+
+stage=0
+cleanup_stage=0
+
+cleanup_affix=cleaned
+nj=100
+
+pad_length=5
+max_silence_length=50
+max_incorrect_words=0
+min_correct_frames=0
+max_utterance_wer=20000
+min_wer_for_splitting=0
+
+ngram_order=2
+
+top_words_interpolation_weight=0.1
```

Comment:
It will be internally normalized to 1. These are the values I used. This way, if someone does not want to use the top words, he can just set it to 0.